dcos_launch/platforms/gce.py

```diff
@@ -227,13 +230,49 @@ def _check_status(response: dict) -> bool:
-        response = self.gce_wrapper.deployment_manager.resources().update(project=self.gce_wrapper.project_id,
-                                                                          deployment=self.name,
-                                                                          body=new_body).execute()
-        log.debug('update response: ' + str(response))
+    def get_resources(self):
+        resources = []
+        request = self.gce_wrapper.deployment_manager.resources().list(project=self.gce_wrapper.project_id,
+                                                                       deployment=self.name)
+        while request is not None:
+            response = request.execute()
+            for resource in response['resources']:
+                resource_copy = copy.deepcopy(resource)
+                for key in resource_copy:
+                    """ The only fields we need as you can see in the templates at the top of this file. Other fields
+                    are present that would cause an error when the API is called for updating a deployment"""
+                    if key not in ('type', 'name', 'metadata', 'properties'):
+                        del resource[key]
+                    """the yaml strings in these fields are not properly formatted for the API (error thrown when
+                    updating a deployment) so we load it and then dump the whole resource data structure correctly in
+                    update_tags afterwards -> default_flow_style=False"""
+                    if key in ['properties', 'metadata']:
+                        resource[key] = yaml.load(resource[key])
+
+                resources.append(resource)
+            request = self.gce_wrapper.deployment_manager.resources().list_next(previous_request=request,
+                                                                                previous_response=response)
+        return {'resources': resources}
+
+    def update_tags(self, tags):
+        info = self.gce_wrapper.deployment_manager.deployments().get(project=self.gce_wrapper.project_id,
+                                                                     deployment=self.name).execute()
+        # we need to get the resources because they're not provided in the info and they're necessary for update
+        info['target'] = {
+            'config': {'content': yaml.dump(self.get_resources(), default_flow_style=False)}
+        }
+        info['labels'] = [{'key': k, 'value': v} for k, v in tags.items()]
+        response = self.gce_wrapper.deployment_manager.deployments().update(project=self.gce_wrapper.project_id,
+                                                                            deployment=self.name, body=info).execute()
+        log.debug('update_tags response: ' + str(response))
         return response
 
+    def get_tags(self):
+        try:
+            return {entry['key']: entry['value'] for entry in self.get_info()['labels']}
+        except KeyError:
```

Comment:
could you either add a comment or make this exception more specific? Right now a keyerror could stem from entry['key'], entry['value'], or self.get_info()['labels']